change_session_id article

SysPete opened the pull request:
I noticed this in the list of ideas so thought I'd make a start.

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,107 @@
+
+It is considered good security practice to change session ID on any change
+of privilege level, for example on login.  It is also a requirement of a
+number of established security standards so is something you should be
```
So *it* is something...

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,107 @@
+=head2 Session destruction
+
+Destroying a session in L<Dancer2> has always been possible using
+L<Dancer2::Core::App/destroy_session> thus:
```
Isn't it "thusly"? Maybe "as such" instead?

SysPete replied:
"thusly" is a colloquial alternative to "thus" in UK English so "thus" is preferred in formal language. Perhaps "like so" would be better.

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,107 @@
+=head2 The gory details
+
+If you are planning to write a new session engine you should really read
+this part...
```
I'd rather have something like "This part is particularly relevant to implementors of session engines".

SysPete replied:
I'll go with that.

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,107 @@
+All session engines should implement the private method C<_change_id>
+which has the signature:
+
+    _change_id $self, $old_id, $new_id
```
No parenthesis?

SysPete commented:
@xsawyerx I've updated (vaguely) along the lines of your review but you should recheck those sections. I also rebased which breaks the "Show outdated" links in the GH page ^^. :unamused:

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,106 @@
+
+L<Dancer2::Core::App/change_session_id> to the rescue!
+
+If you are using a recent version of L<Dancer2::Plugin::Auth::Extensible>
```
I would add in parenthesis which version, like "(starting X.Y.Z)".

SysPete replied:
Agree that it is much better to include the version. Was off-net when I originally wrote this and meant to add in the version before I pushed but forgot. Will reword to say "If you are using version 0.610 or later of..." as that still reads reasonably well and I'm not a big fan of parentheses in prose.

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,106 @@
+    $session_engine->_change_id( $old_id, $new_id )
+
+Where C<$old_id> and C<$new_id>
+are the old and new session IDs respectively. The session engine should do
```
I would break into a new paragraph after "respectively.", since that line is a comment for the code and the rest is explanations.

SysPete replied:
fair enough

xsawyerx commented on pending/change-session-id.pod:
```diff
@@ -0,0 +1,106 @@
+the key used in the session storage.
+
+All file-based session engines such as L<Dancer2::Session::YAML> and
+L<Dancer2::Session::JSON> get native support for C<change_session_id>
```
"such as..." in parenthesis maybe?

xsawyerx commented:
@SysPete, updates?

SysPete commented:
@xsawyerx all changes pushed so unless you still see issues it should be ready to merge.

xsawyerx commented:
:+1: I would like to see this merged. Would you like me to do it?

SysPete commented:
merged via 7ef1f220404539163cd346a8e1b3f7c296a9c0cf
